### PR TITLE
Mct4467

### LINKED ISSUE #4467: [Display Layout] Grid half visible on inspector collapse
Related to :https://github.com/nasa/openmct/issues/4207
#### Summary
Display layout grid is half visible when we toggle inspector collapse.

#### Expected vs Current Behavior
Display layout grid should be reactive

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [ ] Does this impact a critical component?
- [x] Is this just a visual bug?

#### Steps to Reproduce
1. Create or use existing Display Layout.
2. make sure inspector is open and enable edit mode.
3.  now collapse inspector and observe grid

#### Environment
* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Additional Information

Screen Shot 2021-11-15 at 3 09 57 PM

### PR DESCRIPTION
**Pull request by depperm**
Fixes #4467 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [ ] Command line build passes?
* [ ] Has this been smoke tested?
* [ ] Testing instructions included in associated issue?

### CODE REVIEW

File: src/plugins/displayLayout/components/DisplayLayout.vue [hunk 1]
```diff
@@ -233,13 +222,37 @@ export default {
...
+
+                if (item.y + item.height > hMax) {
+                    hMax = item.y + item.height + 2;
+                }
+            });
+            this.gridDimensions = [wMax * this.gridSize[0], hMax * this.gridSize[1]];
+        },
+        watchDisplayResize() {
+            const self = this;
+            const resizeObserver = new ResizeObserver(function () {
```
Comment (jvigliotta) [inline src/plugins/displayLayout/components/DisplayLayout.vue:250]: I think if you use an arrow function here won't need to declare a self variable.

File: src/plugins/displayLayout/components/DisplayLayout.vue [hunk 2]
```diff
@@ -404,7 +417,7 @@ export default {
             }
         },
         containsObject(identifier) {
-            return _.get(this.domainObject, 'composition')
+            return this.domainObject.composition
```
Comment (jvigliotta) [inline src/plugins/displayLayout/components/DisplayLayout.vue:420]: I think if we're replacing `_.get` we may want to check that composition exists (most likely will, but want to be safe), so we can put a check for this.domainObject.composition first and if it doesn't exist, just return;

File: src/plugins/displayLayout/components/DisplayLayout.vue (hunk 1 above)
Comment (depperm) [reply src/plugins/displayLayout/components/DisplayLayout.vue]: fixed

File: src/plugins/displayLayout/components/DisplayLayout.vue (hunk 2 above)
Comment (depperm) [reply src/plugins/displayLayout/components/DisplayLayout.vue]: added a check

**Review by jvigliotta — APPROVED** [review]
### Reviewer Checklist
Check | Passed
-- | --
Changes appear to address issue?? | Y
Appropriate unit tests included? | ??
Code style and in-line documentation are appropriate? | Y
Commit messages meet standards? | Y
Has associated issue been labelled unverified? (if PR closes issue) | Y

**Pull request merged**